indico/web/client/js/jquery/extensions/global.js

```diff
@@ -247,19 +247,26 @@ $(document).ready(function() {
   showFormErrors();
 
   // Show form creation dialog if hash is present in the URL
-  var match = location.hash.match(/^#create-event:(lecture|meeting|conference)(?::(\d+))?$/);
+  let match = location.hash.match(/^#create-event:(lecture|meeting|conference)(?::(\d*)(?::([a-f0-9-]{36}))?)?$/);
   if (match) {
-    var eventType = match[1];
-    var categoryId = match[2];
-    var title = {
+    let eventType = match[1];
+    let categoryId = match[2];
+    let eventUUID = match[3];
+    let title = {
       lecture: $T.gettext('Create new lecture'),
       meeting: $T.gettext('Create new meeting'),
       conference: $T.gettext('Create new conference'),
     }[eventType];
-    var url =
-      match[2] !== undefined
-        ? build_url(Indico.Urls.EventCreation, {event_type: eventType, category_id: categoryId})
-        : build_url(Indico.Urls.EventCreation, {event_type: eventType});
+    let url;
+    if (match[2]) {
+      url = build_url(Indico.Urls.EventCreation, {event_type: eventType, category_id: categoryId});
+    } else if (match[3]) {
+      url = build_url(Indico.Urls.EventCreation, {event_type: eventType, event_uuid: eventUUID});
+    } else if (match[2] && match[3]) {
```

Comment:
this one needs to come first, you never end up in this block right now because the last two if/elif conditions already matched when this one would match as well.

Reply:
bah, github doesn't let me comment on the whole block with a suggestion, so i just paste it here. This is a version with all my comments fixed (and reformatted with prettier)

```js
  const match = location.hash.match(
    /^#create-event:(lecture|meeting|conference)(?::(\d*)(?::([a-f0-9-]{36}))?)?$/
  );
  if (match) {
    const [, eventType, categoryId, eventUUID] = match;
    const title = {
      lecture: $T.gettext('Create new lecture'),
      meeting: $T.gettext('Create new meeting'),
      conference: $T.gettext('Create new conference'),
    }[eventType];
    let url;
    if (categoryId && eventUUID) {
      url = build_url(Indico.Urls.EventCreation, {
        event_type: eventType,
        category_id: categoryId,
        event_uuid: eventUUID,
      });
    } else if (categoryId) {
      url = build_url(Indico.Urls.EventCreation, {event_type: eventType, category_id: categoryId});
    } else if (eventUUID) {
      url = build_url(Indico.Urls.EventCreation, {event_type: eventType, event_uuid: eventUUID});
    } else {
      url = build_url(Indico.Urls.EventCreation, {event_type: eventType});
    }
    ajaxDialog({url, title});
  }
```

Reply:
Thanks, updated.